### PR TITLE
Fix lost adjacent JSX when using Babel

### PR DESCRIPTION
**Pull request by thorn0**
Otherwise Prettier formats `<a/><b/>` to `<a/ >;`.

- [x] I’ve added tests to confirm my change works.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/parser-babylon.js [hunk 1]
```diff
@@ -111,6 +111,20 @@ function tryCombinations(fn, combinations) {
   throw error;
 }
 
+function rethrowSomeRecoveredErrors(ast) {
+  for (const error of ast.errors) {
+    if (
+      typeof error.message === "string" &&
+      error.message.startsWith(
+        "Adjacent JSX elements must be wrapped in an enclosing tag."
```
Comment (thorn0) [inline src/language-js/parser-babylon.js]: @nicolo-ribaudo Would be more convenient if errors had some kind of short codes in order for such checks to not rely on the wording of the message.

**Comment by lydell** [comment]
Ouch! Good find!

This makes me a little bit worried. What other things could Prettier potentially format in a very unexpected way now? Why does Babel even recover from this error?

**Comment by nicolo-ribaudo** [comment]
I'd categorize this as a bug in Babel

**Comment by lydell** [comment]
@nicolo-ribaudo Thanks!

@thorn0 Should we do this the other way around – _whitelist_ what we know is safe, and rethrow everything else? We wouldn’t want Prettier to delete parts of people’s code.

**Comment by nicolo-ribaudo** [comment]
I'm preparing a PR for Babel right now, to not recover from this error. Apart from this everything should be safe (I manually checked a lot of tests, but they were so many that I missed this one).

**Comment by lydell** [comment]
@nicolo-ribaudo Cool, thank you so much! I’ll hold off making the Prettier 1.19 release for a bit then.

**Comment by thorn0** [comment]
Why hold off the release? Let's use my workaround for now.

**Comment by lydell** [comment]
@thorn0 Yeah, if you feel confident about your workaround we could merge this and clean up later after the release.

**Comment by nicolo-ribaudo** [comment]
This workaround looks safe to me.

When do you plan to release 1.19? I can try to get a new Babel version published at most on monday if you prefer.

**Comment by lydell** [comment]
The plan is to release it in the next couple of hours. So let’s merge this PR as-is. Then there’s no rush fixing it on your side.

**Comment by nicolo-ribaudo** [comment]
@lydell Hold off the release, I'm releasing Babel!

**Comment by lydell** [comment]
Alright! :+1:

**Comment by nicolo-ribaudo** [comment]
https://github.com/babel/babel/releases/tag/v7.7.3

**Comment by lydell** [comment]
Yay, thanks for the super quick fix! 🏆 

@thorn0 Would you like to update this PR? Otherwise I can do it.

**Comment by thorn0** [comment]
Yep, I'll do it.

File: yarn.lock [hunk 2]
```diff
@@ -378,7 +378,12 @@
...
     js-tokens "^4.0.0"
 
-"@babel/parser@7.7.2", "@babel/parser@^7.1.2", "@babel/parser@^7.1.5", "@babel/parser@^7.4.5", "@babel/parser@^7.5.5", "@babel/parser@^7.7.0", "@babel/parser@^7.7.2":
+"@babel/parser@7.7.3":
+  version "7.7.3"
+  resolved "https://registry.yarnpkg.com/@babel/parser/-/parser-7.7.3.tgz#5fad457c2529de476a248f75b0f090b3060af043"
+  integrity sha512-bqv+iCo9i+uLVbI0ILzKkvMorqxouI+GbV13ivcARXn9NNEabi2IEz912IgNpT/60BNXac5dgcfjb94NjsF33A==
+
+"@babel/parser@^7.1.2", "@babel/parser@^7.1.5", "@babel/parser@^7.4.5", "@babel/parser@^7.5.5", "@babel/parser@^7.7.0", "@babel/parser@^7.7.2":
   version "7.7.2"
```
Comment (nicolo-ribaudo) [inline yarn.lock]: Is it possible to deduplicate this?

Comment (thorn0) [reply yarn.lock]: done

**Pull request merged**